packages/js/src/ui/components/Header/StatusDropdown.tsx

```diff
@@ -0,0 +1,165 @@
+import { flip, offset, shift } from '@floating-ui/dom';
+import { useFloating, UseFloatingResult } from 'solid-floating-ui';
+import { createSignal, JSX, onCleanup, onMount, Setter, Show } from 'solid-js';
+import { Portal } from 'solid-js/web';
+import { FetchFeedArgs } from 'src/feeds';
+import { NotificationStatus } from 'src/types';
+import { useAppearance, useFeedsContext } from 'src/ui/context';
+import { cn, useStyle } from 'src/ui/helpers';
+import { Archived, ArrowDropDown, Check, Inbox, Unread } from '../../icons';
+import { Dropdown, DropdownItem } from '../common';
+
+const APPEARANCE_KEY_PREFIX = 'inboxStatus';
+
+const DropdownStatus = {
+  UnreadRead: 'Unread & read',
+  Unread: 'Unread only',
+  Archived: 'Archived',
+};
+
+/**
+ *
+ * TODO: Implement getStatusLabel after Filter is implemented
+ * currently, it is a placeholder function
+ */
+const getStatusLabel = (status?: NotificationStatus) => {
+  switch (status) {
+    case NotificationStatus.UNREAD:
+      return 'Inbox';
+    case NotificationStatus.UNSEEN:
+      return 'Unread';
+    case NotificationStatus.SEEN:
+      return 'Archived';
+    default:
+      return 'Inbox';
+  }
+};
+
+export const StatusDropdown = () => {
+  const style = useStyle();
+  const { id } = useAppearance();
+  const [targetRef, setTargetRef] = createSignal<HTMLButtonElement | null>(null);
+  const [contentRef, setContentRef] = createSignal<HTMLDivElement | null>(null);
+  const [isOpen, setIsOpen] = createSignal(false);
+
+  const { setFeedOptions, feedOptions } = useFeedsContext();
+
+  const position = useFloating(targetRef, contentRef, {
+    placement: 'bottom',
+
+    middleware: [
+      offset(8),
+      shift(),
+      flip({
+        fallbackPlacements: ['bottom', 'top'],
+      }),
+    ],
+  });
+
+  const handleClickOutside = (e: any) => {
+    if (contentRef()?.contains(e.target)) return;
+    setIsOpen(false);
+  };
+
+  onMount(() => {
+    document.body.addEventListener('click', handleClickOutside);
```

Comment:
✏️ A good improvement here is to create a custom `useFloating` hook that adds these on and any other functionality we want to have.